src/Microsoft.ML.TorchSharp/NasBert/TextClassificationTrainer.cs

```diff
@@ -0,0 +1,857 @@
+            internal int MaxSequenceLength = 512;
+
+            /// <summary>
+            /// Number of the embedding dimensions. Should be positive. Set by the TorchSharp model.
+            /// </summary>
+            internal int EmbeddingDim = 64;
+
+            /// <summary>
+            /// Number of encoder layers. Set by the TorchSharp model.
+            /// </summary>
+            internal int EncoderLayers = 24;
+
+            /// <summary>
+            ///  Number of the output dimensions of encoder. Should be positive. Set by the TorchSharp model. 3 * EmbeddingDim
+            /// </summary>
+            internal int EncoderOutputDim = 192;
+
+            /// <summary>
+            /// The activation function to use for general situations. Set by the TorchSharp model.
+            /// </summary>
+            internal string ActivationFunction = "gelu";
+
+            /// <summary>
+            /// The activation function to use for pooler layers. Set by the TorchSharp model.
+            /// </summary>
+            internal string PoolerActivationFunction = "tanh";
+
+            /// <summary>
+            /// Reduction of criterion loss function. Set by the TorchSharp model.
+            /// </summary>
+            internal torch.nn.Reduction Reduction = Reduction.Sum;
+        }
+
+        internal TextClassificationTrainer(IHostEnvironment env,
+            string labelColumnName = DefaultColumnNames.Label,
+            string outputColumnName = DefaultColumnNames.PredictedLabel,
+            string sentence1ColumnName = "Sentence1",
+            string sentence2ColumnName = default,
+            int numberOfClasses = 2,
+            int batchSize = 32,
+            int maxEpochs = 10,
+            int maxUpdates = 2147483647,
+            IDataView validationSet = null,
+            TextClassificationTrainer.Architecture architecture = Architecture.Roberta) :
+            this(env, new Options
+            {
+                OutputColumnName = outputColumnName,
+                Sentence1ColumnName = sentence1ColumnName,
+                Sentence2ColumnName = sentence2ColumnName,
+                LabelColumnName = labelColumnName,
+                BatchSize = batchSize,
+                NumberOfClasses = numberOfClasses,
+                MaxEpoch = maxEpochs,
+                MaxUpdate = maxUpdates,
+                ValidationSet = validationSet,
+            })
+        {
+        }
+
+        internal TextClassificationTrainer(IHostEnvironment env, Options options)
+        {
+            _host = Contracts.CheckRef(env, nameof(env)).Register(nameof(TextClassificationTrainer));
+            _options = options;
+        }
+
+        public SentenceClassificationTransformer Fit(IDataView input)
+        {
+            using (var ch = _host.Start("TrainModel"))
+            using (var pch = _host.StartProgressChannel("Training model"))
+            {
+                var header = new ProgressHeader(new[] { "Total Terms" }, null);
```

Comment:
>new[] { "Total Terms" } [](http://example.com/codeflow?start=48&length=23)

can this be a static? or it is expected to mutated later?

Reply:
I don't think it will be mutated? This is just how we have done it in the past so its how I did it this time. This code will only happen once per trainer though, so its not on a hot path.